src/schema/autohost/installEngine.ts

```diff
@@ -0,0 +1,33 @@
+import { Type } from "@sinclair/typebox";
+
+import { defineEndpoint } from "@/generator-helpers.js";
+
+export default defineEndpoint({
+    source: "server",
+    target: "autohost",
+    description: "Ask the autohost to install specified engine version.",
+    request: {
+        data: Type.Object(
+            {
+                version: Type.String({ description: "Version of the engine to install" }),
+            },
+            {
+                examples: [
+                    {
+                        version: "2025.01.5",
+                    },
+                ],
+            }
+        ),
+    },
+    response: [
+        {
+            status: "success",
```

Comment:
What does success mean here? request received, engine installed, something else?

Reply:
Installed, I've added the description.

Reply:
How long do you reckon that operation could take? I’m thinking I may need a higher timeout for response on that one maybe?

Reply:
Hard to tell, I will have some internal timeouts and retries, they will be probably max ~5min, because download from GitHub can be very very slow in some cases. In normal situations it should be < 15s

Would you prefer to instead have just "trigger download" or something like that?

I don't like long running rpcs too...

Reply:
If you want we can add `deadline` argument and I will ensure return before deadline is reached.

Reply:
not sure about the deadline bit, seems it’s just going to add complexity for not much benefit.
If autohost gets another request to install an engine while it’s already dealing with the same request, what should happen?
I can increase the timeout and do some retries, and in this case it would be good to know how to handle that use case.

Reply:
If you send another request for the engine version that was already requested to be installed via a previous request, I will dedup it internally.

So it's really a question for you:
- Do you want instant response
- Or do you want possibly long response but also potential report with error (not sure of you need it anyway)

The available versions are still distributed via the status update, so it's effectively a question if you want/need to track or something I dunning.

For me, for autohost implementation, it doesn't matter at all which option you pick.

Reply:
I'd say let’s make this a request to do things in the background, and "success" means the autohost acknoledged it, but may not have completed it.

Reply:
Okey, pushed the change.